### PR TITLE
Bug/original image

### PR DESCRIPTION
**Pull request by epureanudiana**

### CODE REVIEW

File: utils/dataloaders.py [hunk 1]
```diff
@@ -621,16 +621,17 @@ def __init__(self,
...
             results = ThreadPool(NUM_THREADS).imap(fcn, range(n))
             pbar = tqdm(enumerate(results), total=n, bar_format=TQDM_BAR_FORMAT, disable=LOCAL_RANK > 0)
             for i, x in pbar:
                 if cache_images == 'disk':
                     b += self.npy_files[i].stat().st_size
                 else:  # 'ram'
-                    self.ims[i], self.im_hw0[i], self.im_hw[i] = x  # im, hw_orig, hw_resized = load_image(self, i)
+                    self.ims[i], self.im_hw0[i], self.im_hw[i], self.ims_orig[i] = x  # im, hw_orig, hw_resized, im_orig = load_image(self, i)
                     b += self.ims[i].nbytes
+                    b += self.ims_orig[i].nbytes
```
Comment (sdavrieux) [inline utils/dataloaders.py:634]: Assuming that I don't know this code very well. Does it makes sense to sum both the bytes of ims and ims_orig. Or should be one or the other?

File: utils/dataloaders.py [hunk 2]
```diff
@@ -704,18 +705,21 @@ def __getitem__(self, index):
...
             shapes = None
 
             # MixUp augmentation
             if random.random() < hyp['mixup']:
                 img, labels = mixup(img, labels, *self.load_mosaic(random.randint(0, self.n - 1)))
-
         else:
             # Load image
-            img, (h0, w0), (h, w) = self.load_image(index)
-            im0 = img.copy()
```
Comment (sdavrieux) [inline utils/dataloaders.py:718]: So im0 is gone?

File: utils/dataloaders.py [hunk 3]
```diff
@@ -782,12 +786,13 @@ def load_image(self, i):
...
                 assert im is not None, f'Image Not Found {f}'
             h0, w0 = im.shape[:2]  # orig hw
+            im_orig = np.copy(im)
             r = self.img_size / max(h0, w0)  # ratio
             if r != 1:  # if sizes are not equal
                 interp = cv2.INTER_LINEAR if (self.augment or r > 1) else cv2.INTER_AREA
                 im = cv2.resize(im, (math.ceil(w0 * r), math.ceil(h0 * r)), interpolation=interp)
-            return im, (h0, w0), im.shape[:2]  # im, hw_original, hw_resized
-        return self.ims[i], self.im_hw0[i], self.im_hw[i]  # im, hw_original, hw_resized
+            return im, (h0, w0), im.shape[:2], im_orig  # im, hw_original, hw_resized
```
Comment (sdavrieux) [inline utils/dataloaders.py:794]: I don't really get these comments repeating what's already written in code, but anyway, `im_orig` is missing.

File: utils/plots.py [hunk 4]
```diff
@@ -295,7 +295,7 @@ def plot_images(images, targets, paths=None, fname='images.jpg', names=None):
                 cls = classes[j]
                 color = colors(cls)
                 cls = names[cls] if names else cls
-                if labels or conf[j] > 0.25:  # 0.25 conf thresh
+                if labels or conf[j] > conf_thres:  # 0.25 conf thresh
```
Comment (sdavrieux) [inline utils/plots.py:298]: We can remove this comment since it's not always 0.25

File: utils/dataloaders.py (hunk 1 above)
Comment (epureanudiana) [reply utils/dataloaders.py:637]: im_orig is now also added in memory, so it makes sense to be added

File: utils/dataloaders.py (hunk 2 above)
Comment (epureanudiana) [reply utils/dataloaders.py:718]: yes, replaced by im_orig

File: val.py [hunk 5]
```diff
@@ -180,17 +181,15 @@ def run(
         callbacks=Callbacks(),
         compute_loss=None,
         tagged_data=False,
-        skip_evaluation=True,
+        skip_evaluation=False,
```
[the remark this answers is not in the document]
Comment (epureanudiana) [reply val.py:190]: Yes, im sure. This boolean is now set at pipelines level and it tested both of them

File: utils/dataloaders.py (hunk 3 above)
Comment (epureanudiana) [reply utils/dataloaders.py]: ill remove them

File: val.py [hunk 6]
```diff
@@ -441,17 +438,16 @@ def run(
...
-
+                pred_clone[:, :4] = scale_boxes(im[si].shape[1:], pred_clone[:, :4], shape, shapes[si][1])
                 for *xyxy, conf, cls in pred_clone.tolist():
                     x1, y1 = int(xyxy[0]), int(xyxy[1])
                     x2, y2 = int(xyxy[2]), int(xyxy[3])
-                    area_to_blur = im0[si][y1:y2, x1:x2]
-
+                    if x1 == x2 or y1 == y2:
+                        LOGGER.warning('Area to blur is 0.')
+                        continue
```
[the remark this answers is not in the document]
Comment (epureanudiana) [reply val.py]: I'll reformat! thanks

**Review by epureanudiana — COMMENTED** [review]
Thanks for the comments!

File: utils/dataloaders.py (hunk 1 above)
Comment (sdavrieux) [reply utils/dataloaders.py:637]: Oooh `b` are the bytes in memory? 
The naming of things in this project is awful I have to say.

**Review by sdavrieux — APPROVED** [review]

**Pull request merged**